custom_components/waste_collection_schedule/waste_collection_schedule/source/dillingen_saar_de.py

```diff
@@ -0,0 +1,93 @@
+import datetime
+import requests
+from waste_collection_schedule import Collection
+from waste_collection_schedule.service.ICS import ICS
+from urllib.parse import quote
+from datetime import datetime
+
+
+
+TITLE = "Dillingen Saar" # Title will show up in README.md and info.md
+DESCRIPTION = "Source script for waste collection Dillingen Saar"  # Describe your source
+URL = "https://www.dillingen-saar.de/"  # Insert url to service homepage. URL will show up in README.md and info.md
+TEST_CASES = {  # Insert arguments for test cases to be used by test_sources.py script
+    "Am Fischerberg": {"street": "Am Fischerberg"}, #https://service-dillingen-saar.fbo.de/date/Am%20Fischerberg/2023-01-01/+1%20year/?format=ics&type=rm,gs,bio,pa
+    "Odilienplatz": {"street": "Odilienplatz"},     #https://service-dillingen-saar.fbo.de/date/Odilienplatz/2023-01-01/+1%20year/?format=ics&type=rm,gs,bio,pa
+    "Lilienstraße": {"street": "Lilienstraße"},      #https://service-dillingen-saar.fbo.de/date/Lilienstra%C3%9Fe/2023-01-01/+1%20year/?format=ics&type=rm,gs,bio,pa
+    "Joseph-Roederer-Straße": {"street": "Joseph-Roederer-Straße"} #https://service-dillingen-saar.fbo.de/date/Joseph-Roederer-Stra%C3%9Fe/2023-01-01/+1%20year/?format=ics&type=rm,gs,bio,pa
+}
+
+API_URL = "https://service-dillingen-saar.fbo.de/date/"
+REMINDER_DAY = (
+    "0"  # The calendar event should be on the same day as the waste collection
+)
+REMINDER_TIME = "0600"  # The calendar event should start on any hour of the correct day, so this does not matter much
+ICON_MAP = {
+    "Altpapier": "mdi:package-variant",
+    "Restmüll": "mdi:trash-can",
+    "Gelber Sack": "mdi:recycle",
+}
+
+SPECIAL_CHARS = str.maketrans(
+    {
+        " ": "%20",
+        "ä": "ae",
+        "ß": "%C3%9F",
+        "ü": "ue",
+        "ö": "oe",
+        "ß": "ss",
+        "(": None,
+        ")": None,
+        ",": None,
+        ".": None,
+    }
+)
+
+class Source:
+    def __init__(self, street: str):  # argX correspond to the args dict in the source configuration
+#        self._street = quote(
+#            street.lower().translate(SPECIAL_CHARS).strip()
+#        ) 
+        self._street = street
+        self._ics = ICS()
+#LATER        self._ics = ICS(regex=r" ...\n (.*)", split_at=r" & ") # remove trailing ...\n from SUMMARY field
+        # 'r' at the start of the pattern string designates a python "raw" string which passes through backslashes without change
+
+
+    def fetch(self):
+
+        # the url contains the current year, but this doesn't really seems to matter at least for the ical, since the result is always the same
+        # still replace it for compatibility sake
+        now = datetime.now()
+
+        url = f"{API_URL}/{self._street}/{str(now.year)}-01-01/+1%20year/?format=ics&type=rm,gs,bio,pa"
+        res = requests.get(url)
+        res.raise_for_status()
+
+
+
+        # Fix issue with a mix of Windows/Mac line breaks
+        ics_data = res.text.replace("\r", "\n")
```

Comment:
Do you really need these 2 replace statements? I'm wondering because I have never heard of an issue with the ICS parser with Windows/Mac line breaks...